user_tools/src/spark_rapids_pytools/cloud_api/dataproc.py

```diff
@@ -161,15 +161,20 @@ def get_matching_executor_instance(self, cores_per_executor):
         # If the number of cores is a valid entry in the list, return the equivalent instance
         if cores_per_executor in unit_info['vCPUs']:
             return f'{series_name}-{cores_per_executor}'
-        # If the num-cores is not in the list, then we need to adjust the cores to the best match.
-        # The loop below should always return a valid instance type. It takes the instance type with
-        # number of cores less than or equal to the "cores_per_executor".
-        adjusted_cores = unit_info['vCPUs'][0]
-        for num_cpu in unit_info['vCPUs'][1:]:
-            if num_cpu <= cores_per_executor:
-                adjusted_cores = num_cpu
-        # At this point adjusted_cores should be the best match. The abobe code
-        return f'{series_name}-{adjusted_cores}'
+        # If the num-cores is not in the list, then we need to adjust the cores to the best match
+        # that is greater than the input cores.
+        adjusted_cores = unit_info['vCPUs'][-1]
+        for arr_index in range(len(unit_info['vCPUs']) - 1, -1, -1):
+            # do not break the loop just in case the array is not sorted
```

Comment:
we should probably sort this since it could get wrong number  if say you have [32, 24, 16, 8] and you are trying to match 12.  it would end up with 32 instead of 16.